### PR TITLE
Import rides in SV6 files per ride and per field

### PR DESCRIPTION
**Pull request by Gymnasiast**
Another small step in moving away from SV6.
Further clean-up is possible by grouping loops with the same maximum together,
but this is easier to check.

### CODE REVIEW

**Comment by IntelOrca** [comment]
And big-endian.

File: src/openrct2/rct2/S6Importer.cpp [hunk 1]
```diff
@@ -348,7 +348,9 @@ class S6Importer final : public IParkImporter
         memcpy(gBanners, _s6.banners, sizeof(_s6.banners));
         memcpy(gUserStrings, _s6.custom_strings, sizeof(_s6.custom_strings));
         gCurrentTicks = _s6.game_ticks_1;
-        memcpy(gRideList, _s6.rides, sizeof(_s6.rides));
+        //memcpy(gRideList, _s6.rides, sizeof(_s6.rides));
```
Comment (IntelOrca) [inline src/openrct2/rct2/S6Importer.cpp]: You can remove this.

File: src/openrct2/rct2/S6Importer.cpp [hunk 2]
```diff
@@ -425,6 +427,237 @@ class S6Importer final : public IParkImporter
...
     }
 
+    void ImportRides()
+    {
+        for (uint16 index = 0; index < RCT2_MAX_RIDES_IN_PARK; index++)
+        {
+            if (_s6.rides[index].type != RIDE_TYPE_NULL)
+            {
+                ImportRide(get_ride(index), &_s6.rides[index]);
+
```
Comment (IntelOrca) [inline src/openrct2/rct2/S6Importer.cpp]: Why the blank line?

Comment (Gymnasiast) [inline src/openrct2/rct2/S6Importer.cpp]: Just a typo.

Comment (IntelOrca) [inline src/openrct2/rct2/S6Importer.cpp]: There is another below.

**Comment by Gymnasiast** [comment]
Ok, that should your line notes be addressed.

**Comment by janisozaur** [comment]
@Gymnasiast are you going to do the same thing for export?

File: src/openrct2/rct2/S6Importer.cpp [hunk 3]
```diff
@@ -425,6 +427,235 @@ class S6Importer final : public IParkImporter
...
+        for (uint16 index = 0; index < RCT2_MAX_RIDES_IN_PARK; index++)
+        {
+            if (_s6.rides[index].type != RIDE_TYPE_NULL)
+            {
+                ImportRide(get_ride(index), &_s6.rides[index]);
+            }
+        }
+    }
+
+    void ImportRide(rct_ride * dst, rct_ride * src)
```
Comment (janisozaur) [inline src/openrct2/rct2/S6Importer.cpp]: `const rct_ride * src`

**Comment by IntelOrca** [comment]
@janisozaur I don't think it's worth it, it will be removed when we have our own format.

File: src/openrct2/rct2/S6Importer.cpp (hunk 3 above)
Comment (Gymnasiast) [inline src/openrct2/rct2/S6Importer.cpp]: Changed.

**Pull request merged**